### PR TITLE
Add og:images generation

### PR DESCRIPTION
**Pull request by teleginzhenya**
* separate page for og:images preview
* pass custom meta for libraries
* script which runs puppeteer and makes screenshots
* github action which commits it and push to repositry

design of og:images not final

### CODE REVIEW

**Comment by teleginzhenya** [comment]
not sure how gh-actions will work :]
but main idea to create images, commit them and then run deployment

File: public/og-images/app-layout.jpg
Comment (melikhov-dev) [inline public/og-images/app-layout.jpg:1]: From my point of view text is so tiny

File: scripts/prepare-og-images.mjs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+try {
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
+
+    for (let i = 0; i < libs.length; i++) {
+        const { id } = libs[i];
+        await page.goto(`http://localhost:3000/preview/${id}`, { waitUntil: 'networkidle0' });
+        await page.setViewport({ width: 1200, height: 630 });
+        await page.screenshot({ path: `public/og-images/${id}.jpg` })
```
Comment (melikhov-dev) [inline scripts/prepare-og-images.mjs:17]: I understand that, but I don't like it. :)

File: next.config.js [hunk 2]
```diff
@@ -62,4 +62,11 @@ const plugins = [
 module.exports = withPlugins(plugins, {
     reactStrictMode: true,
     output: 'export',
+    async redirects() {
```
Comment (imsitnikov) [inline next.config.js:65]: Does `redirects` method work after `next export`?

File: src/components/Layout/Meta/Meta.tsx [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+    description: 'Build modern interfaces with the Gravity design system and libraries',
+    image: 'https://gravity-ui.com/index-social.png',
+};
+
+export type MetaProps = {
+    name?: string;
+    description?: string;
+    image?: string;
+};
+export const Meta: React.FC = ({
```
Comment (imsitnikov) [inline src/components/Layout/Meta/Meta.tsx:14]: `React.FC<MetaProps>` – I have lint errors without that

File: src/components/Layout/Meta/Meta.tsx [hunk 4]
```diff
@@ -1,6 +1,21 @@
 import React from 'react';
 
-export const Meta: React.FC = () => {
+const DEFAULT_META = {
+    name: 'Gravity&nbsp;UI',
+    description: 'Build modern interfaces with the Gravity design system and libraries',
+    image: 'https://gravity-ui.com/index-social.png',
+};
+
+export type MetaProps = {
```
Comment (imsitnikov) [inline src/components/Layout/Meta/Meta.tsx:9]: Add new line after `MetaProps` plz

File: package.json [hunk 5]
```diff
@@ -41,6 +41,7 @@
     "next-compose-plugins": "^2.2.1",
     "next-transpile-modules": "^10.0.0",
     "prettier": "^2.7.1",
+    "puppeteer": "^19.10.1",
```
Comment (imsitnikov) [inline package.json:44]: Can we use the playwright (https://www.npmjs.com/package/playwright) here? We've just moved from puppeeter to playwright in other projects)

File: public/og-images/app-layout.jpg
Comment (teleginzhenya) [reply public/og-images/app-layout.jpg:1]: design of og:images not final

File: next.config.js (hunk 2 above)
Comment (teleginzhenya) [reply next.config.js]: I've removed this part in favour of getStaticProps https://github.com/gravity-ui/landing/pull/45/commits/cc9e761e8c7266543d3fe72de983968a1742572a

File: src/components/Layout/Layout.tsx [hunk 6]
```diff
@@ -28,12 +29,12 @@ const theme = Theme.Dark;
 configureUiKit({lang});
 configurePageConstructor({lang});
 
-export const Layout: React.FC<LayoutProps> = ({title, children}) => {
+export const Layout: React.FC<LayoutProps> = ({title, children, meta}) => {
```
[the remark this answers is not in the document]
Comment (teleginzhenya) [reply src/components/Layout/Layout.tsx]: 46c7309c62ff9357753d5b4096b099d7e2b3f590

File: package.json (hunk 5 above)
Comment (teleginzhenya) [reply package.json]: 94fcdba80ad7e467d8ecc2e4a50e131d84e0fa25

File: scripts/prepare-og-images.mjs (hunk 1 above)
Comment (teleginzhenya) [reply scripts/prepare-og-images.mjs]: 94fcdba80ad7e467d8ecc2e4a50e131d84e0fa25 moved to function :D

Comment (melikhov-dev) [reply scripts/prepare-og-images.mjs]: nope, I'm talking about using headless browser :)

**Comment by lxndr** [comment]
Next.js 14 got support for open graph images (https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image). Has anyone considered using that?

**Comment by teleginzhenya** [comment]
closed in favor of #236

**Pull request closed without merging**